audio-worklet/src/noisegate-script-processor.js

```diff
@@ -172,24 +180,47 @@ class NoiseGate {
     // releasing (between closed and open).
     for (let i = 0; i < envelope.length; i++) {
       if (envelope[i] < this.threshold) {
-        const weight = this.lastWeight_ - attackLossPerStep;
+        const weight = this.previousWeight_ - attackLossPerStep;
         this.weights_[i] = Math.max(weight, 0);
       }
       else {
-        const weight = this.lastWeight_ + releaseGainPerStep;
+        const weight = this.previousWeight_ + releaseGainPerStep;
         this.weights_[i] = Math.min(weight, 1);
       }
-      this.lastWeight_ = this.weights_[i];
+      this.previousWeight_ = this.weights_[i];
     }
     return this.weights_;
   }
 
-  getFilterCoefficient_(timeConstant) {
-    let alpha = Math.exp(-1 / (this.context_.sampleRate * timeConstant));
+  /**
+   * Sets the envelope's alpha weight.
```

Comment:
I think this comment is misleading.  Just say this computes the filter coefficient for the envelope filter.